mlflow/xgboost.py

```diff
@@ -514,9 +522,19 @@ def log_feature_importance_plot(features, importance, importance_type):
             )
             if early_stopping:
                 extra_step = len(eval_results)
-                metrics_logger.record_metrics({"stopped_iteration": extra_step - 1})
-                metrics_logger.record_metrics({"best_iteration": model.best_iteration})
-                metrics_logger.record_metrics(eval_results[model.best_iteration], extra_step)
```

Comment:
`BatchMetricsLogger` is tailored to per-epoch metric logging use cases from callbacks; it may make multiple logging API calls during this segment. The new `MlflowAutologgingQueueingClient` is a better choice here; it will only make a single API call to LogBatch.